### DISCUSSION TITLE
Lesson 7: Smart contract lottery

### DISCUSSION THREAD

**Discussion by arifulone** [smartcontractkit/full-blockchain-solidity-course-py#1768] [Q&A] [answered]
VRF v1  doesn't exist anymore .  How do i work? 
there is no guide about VRF v2  
plz Help Me out

**Comment by cromewar** [comment, marked as answer]
Hello @arf92 
Yes you are right VRF1 was deprecated. so please follow this guides:
1. Official Chainlink VRF2 video: https://youtu.be/rdJ5d8j1RCg.
2. Take a look to Chainlink Documentation: https://docs.chain.link/docs/vrf/v2/introduction/

If you have any extra information, please let us know we will try to provide you more support.

**Comment by Niferu** [comment]
Hello @arf92
As far as I know you can still do everything from course using VRF v1 as it works for me. You just have to use correct addresses for current test network, which is goerli. You can find everything here: https://docs.chain.link/docs/vrf/v1/supported-networks/

**Reply by cromewar** [reply]
This is also true, but the official documentation for VR1 is also getting deprecated so learning V2 is also a good idea.

**Reply by Niferu** [reply]
Indeed it is, but implementing it into code if you are totally green isn't easy at all, so I would stay with v1 until I learn more ;)
